v2/src/log_context_property_basic_types.c

```diff
@@ -13,149 +14,209 @@
-        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_070: [ LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).get_type shall returns the property type LOG_CONTEXT_PROPERTY_TYPE_int8_t. ]*/ \
-        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_080: [ LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint8_t).get_type shall returns the property type LOG_CONTEXT_PROPERTY_TYPE_uint8_t. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_010: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).get_type shall return the property type LOG_CONTEXT_PROPERTY_TYPE_int64_t. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_020: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(uint64_t).get_type shall return the property type LOG_CONTEXT_PROPERTY_TYPE_uint64_t. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_030: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int32_t).get_type shall return the property type LOG_CONTEXT_PROPERTY_TYPE_int32_t. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_040: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(uint32_t).get_type shall return the property type LOG_CONTEXT_PROPERTY_TYPE_uint32_t. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_050: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int16_t).get_type shall return the property type LOG_CONTEXT_PROPERTY_TYPE_int16_t. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_060: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(uint16_t).get_type shall return the property type LOG_CONTEXT_PROPERTY_TYPE_uint16_t. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_070: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int8_t).get_type shall return the property type LOG_CONTEXT_PROPERTY_TYPE_int8_t. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_080: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(uint8_t).get_type shall return the property type LOG_CONTEXT_PROPERTY_TYPE_uint8_t. ]*/ \
         return MU_C2(LOG_CONTEXT_PROPERTY_TYPE_, type_name); \
     } \
 
 #define DEFINE_BASIC_TYPE_IF(type_name) \
-    const LOG_CONTEXT_PROPERTY_TYPE_IF MU_C2(type_name, _log_context_property_type) = \
+    const LOG_CONTEXT_PROPERTY_TYPE_IF LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(type_name) = \
     { \
-        .to_string = MU_C2(type_name, _log_context_property_to_string), \
-        .copy = MU_C2(type_name, _log_context_property_copy), \
-        .free = MU_C2(type_name, _log_context_property_free), \
-        .get_type = MU_C2(type_name, _log_context_property_get_type) \
+        .to_string = MU_C2(type_name, _log_context_property_type_to_string), \
+        .copy = MU_C2(type_name, _log_context_property_type_copy), \
+        .free = MU_C2(type_name, _log_context_property_type_free), \
+        .get_type = MU_C2(type_name, _log_context_property_type_get_type) \
     }; \
 
+#define DEFINE_BASIC_TYPE_INIT(type_name, print_format_string) \
```

Comment:
>print_format_string [](http://example.com/codeflow?start=42&length=19)

except for printing it, it is not really used...

Is it needed?

Reply:
Correct, we use it for printing.